framework/src/controls/PIDTransientControl.C

```diff
@@ -109,14 +111,30 @@ PIDTransientControl::PIDTransientControl(const InputParameters & parameters)
 void
 PIDTransientControl::execute()
 {
+  // Dont execute on INITIAL again on a recover.
+  // Executing on INITIAL in a regular simulation is fine, but on a recover we will get the integral
+  // term wrong and set the wrong attributes for the time derivative too. Best to skip
+  if (_app.isRecovering() && _fe_problem.getCurrentExecuteOnFlag() == EXEC_INITIAL)
+  {
+    mooseInfo("Skipping execution on recover + INITIAL.");
+    return;
+  }
+
   if (_t >= _start_time && _t < _stop_time)
   {
     // Get the current value of the controllable parameter
-    Real value;
     if (isParamValid("parameter"))
-      value = getControllableValue<Real>("parameter");
+    {
+      // If just recovering, we must use the restartable value as the parameter value is
+      // not restarted
+      if (_app.isRecovering() && !_has_recovered)
+        _has_recovered = true;
```

Comment:
This flag needs to be reset to false at some point to have the same behavior for subsequent recovers.

Reply:
it's not restartable so it should reset itself on every recover